bot/exts/evergreen/youtube.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from dataclasses import dataclass
+from html import unescape
+from typing import List, Optional
+from urllib.parse import quote_plus
+
+from discord import Embed
+from discord.ext import commands
+from discord.utils import escape_markdown
+
+from bot.constants import Colours, Emojis, Tokens
+
+log = logging.getLogger(__name__)
+
+KEY = Tokens.youtube
+SEARCH_API = "https://www.googleapis.com/youtube/v3/search"
+STATS_API = "https://www.googleapis.com/youtube/v3/videos"
+YOUTUBE_VIDEO_URL = "https://www.youtube.com/watch?v={id}"
+YOUTUBE_SEARCH_URL = "https://www.youtube.com/results?search_query={search}"
+RESULT = (
+    "**{index}. [{title}]({url})**\n"
+    "{post_detail_emoji} {user_emoji} {username} {view_emoji} {view_count} {like_emoji} {like_count}\n"
+)
+
+
+@dataclass
+class VideoStatistics:
+    """Represents YouTube video statistics."""
+
+    view_count: int
+    like_count: int
+
+
+@dataclass
+class Video:
+    """Represents a video search result."""
+
+    title: str
+    username: str
+    id: str
+    video_statistics: VideoStatistics
+
+
+class YouTubeSearch(commands.Cog):
+    """Sends the top 5 results of a query from YouTube."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    async def format_search_result(self, index: int, result: List[Video]) -> str:
+        """Formats search result to put in embed."""
+        return RESULT.format(
```

Comment:
Maybe this is just my personal bias, but I think it's hard to read formatting code if the formatting template is in a different castle. Maybe put it nearby, or even inline it (as an f-string?)

Reply:
Does this include the `SEARCH_API`, `STATS_API`, `YOUTUBE_VIDEO_URL`, and `YOUTUBE_SEARCH_URL` constants too? Or are those fine as is considering that their names are self-explanatory compared to the `RESULT` f-string.